modules/@apostrophecms/modal/ui/apos/components/AposModalShareDraft.vue

```diff
@@ -79,22 +86,89 @@ export default {
         disableHeader: true,
         trapFocus: true
       },
-      disabled: false
+      shareUrl: '',
+      disabled: true
     };
   },
   async mounted() {
     this.modal.active = true;
+    this.checkUrlprop();
   },
   methods: {
-    copy() {
-      console.log('copy');
+    async copy() {
+      await navigator.clipboard.writeText(this.shareUrl);
     },
-    toggle() {
-      console.log('toggle');
+    async toggle() {
       this.disabled = !this.disabled;
+
+      await this.setShareUrl();
+    },
+    async setShareUrl() {
+      try {
+        const { aposShareKey } = await apos.http.post(
+          `${apos.modules[this.doc.type].action}/${this.doc._id}/share`, {
+            busy: true,
+            body: {
+              share: !this.disabled
+            },
+            draft: true
+          }
+        );
+
+        if (this.disabled) {
+          this.shareUrl = '';
+          return;
+        }
+
+        if (!aposShareKey) {
+          return this.errorNotif();
+        }
+
+        this.shareUrl = this.generateShareUrl(aposShareKey);
+
+      } catch (err) {
+        if (this.disabled) {
+          this.shareUrl = '';
+        } else {
+          this.errorNotif();
+        }
+      }
     },
     close() {
       this.modal.showModal = false;
+    },
+    async checkUrlprop() {
+      if (!this.doc._url) {
+        await this.errorNotif();
+      } else if (this.doc.aposShareKey) {
+        this.disabled = false;
+        this.shareUrl = this.generateShareUrl(this.doc.aposShareKey);
+      }
+    },
+    async errorNotif() {
+      await apos.notify('apostrophe:shareDraftError', {
+        type: 'danger',
+        icon: 'alert-circle-icon',
+        dismiss: true
+      });
+
+      setTimeout(() => {
+        this.close();
+      }, 500);
+    },
+    generateShareUrl(aposShareKey) {
+      const qs = Object.fromEntries(
+        Object.entries(apos.http.parseQuery(this.doc._url))
+          .filter(([ _, val ]) => val)
+      );
+
+      const slug = apos.http.addQueryToUrl(this.doc._url, {
+        ...qs,
```

Comment:
```suggestion
```

if `this.doc._url` doesn't contain any query string, see https://github.com/apostrophecms/apostrophe/pull/3793/files#r893504423

Reply:
Maybe for the pages?